botocore/signers.py

```diff
@@ -196,6 +208,65 @@ def generate_presigned_url(self, request_dict, expires_in=3600,
         return request.url
 
 
+class CloudFrontSigner(object):
+
+    @staticmethod
```

Comment:
I think it makes sense to remove this method and merge it into the signing method. This method is not just base64 encoding a string, but also transforming the string to conform with the unique requirements of CloudFront urls. This is not really something that needs to be exposed publicly.